tests/components/concentration_tracker/test_concentration_tracker_for_diffusion.py

```diff
@@ -0,0 +1,281 @@
+"""
+Created on Wed Jul 12 12:25:27 2023
+
+@author: LaurentRoberge
+"""
+
+import numpy as np
+import pytest
+
+from landlab import FieldError, RasterModelGrid
+from landlab.components import ConcentrationTrackerForDiffusion
+
+# %% Test input field errors
+
+
+def test_input_soil_flux_from_diffuser():
+    """
+    ConcentrationTrackerForDiffusion should throw an error when output fields
+    from a diffusion component do not exist (soil__flux)
+    """
+    # Make a raster model grid
+    mg = RasterModelGrid((3, 3))
+    _ = mg.add_zeros("soil__depth", at="node")
+    _ = mg.add_zeros("soil_production__rate", at="node")
+    _ = mg.add_zeros("topographic__elevation", at="node")
+
+    # Instantiate the component
+    with pytest.raises(FieldError):
+        _ = ConcentrationTrackerForDiffusion(mg)
+
+
+def test_input_fields_soil():
+    """
+    ConcentrationTrackerForDiffusion should throw an error when input fields
+    are not provided (soil__depth, soil_production__rate, topographic__elevation)
+    """
+    mg = RasterModelGrid((3, 3))
+    _ = mg.add_zeros("soil__flux", at="link")
+
+    fields_to_add = ["soil__depth", "soil_production__rate", "topographic__elevation"]
+
+    # Instantiate the component
+    for field in fields_to_add:
+        with pytest.raises(FieldError):
+            _ = ConcentrationTrackerForDiffusion(mg)
+        _ = mg.add_zeros(field, at="node")
```

Comment:
This would be a good place to use a parameterized test.
```python
@pytest.mark.parametrize(
    "required_field", ["soil__depth", "soil_production__rate", "topographic__elevation"]
)
def test_input_fields_soil(required_field):
    """
    ConcentrationTrackerForDiffusion should throw an error when input fields
    are not provided (soil__depth, soil_production__rate, topographic__elevation)
    """
    mg = RasterModelGrid((3, 3))
    mg.add_zeros("soil__flux", at="link")

    mg.add_zeros("soil__depth", at="node")
    mg.add_zeros("soil_production__rate", at="node")
    mg.add_zeros("topographic__elevation", at="node")

    mg.at_node.pop(required_field)
    with pytest.raises(FieldError):
        ConcentrationTrackerForDiffusion(mg)
```